Add note on yaml types (#3021)
withastro/docs: src/content/docs/en/recipes/add-yaml-support.mdx

    ```js
    import yml from './data.yml';
    ```

    :::note

>     While you can now import YAML data in your Astro project, your editor will not provide types for the imported data. To add types, create a `*.d.ts` file in your project and add the following:

Author: ```suggestion
    While you can now import YAML data in your Astro project, your editor will not provide types for the imported data. To add types, create a `*.d.ts` file in the `src` directory of your project and add the following:
```

>     While you can now import YAML data in your Astro project, your editor will not provide types for the imported data. To add types, create a `*.d.ts` file in the `src` directory of your project and add the following:

Reviewer: ```suggestion
    While you can now import YAML data in your Astro project, your editor will not provide types for the imported data. To add types, create or find an existing `*.d.ts` file in the `src` directory of your project and add the following:
```

Also just to check:
- Must this go in its own, new file? Can it go in an existing file along with other things? If so, I might suggest something like above.

Author: Yep, an existing file is fine!

>     ```ts

Reviewer: ```suggestion
    ```ts title="src/*.d.ts"
```
Is there any reason not to do this? (Though, if `*` is a wildcard, I'd prefer an actual file name here someone might use.)

Author: We could set it to `files.d.ts` to show it's intentions?
```suggestion
    ```ts title="src/files.d.ts"
```

    // Specify the file extension you want to import
    declare module "*.yml" {

>       const value: any; // Add better type definitions here if desired

Reviewer: ```suggestion
      const value: any; // Add extra type definitions here if desired
```
Can we go for a different word for "better"?  :sweat_smile: I tried "extra" here, but you could say something like "additional" or "Add any desired type definitions here"

Author: Could removing the word entirely work? Since `any` isn't really a type, and is just a placeholder.
```suggestion
      const value: any; // Add type definitions here if desired
```

If not "additional" sounds like it works

```suggestion
      const value: any; // Add additional type definitions here if desired
```

>     ```ts
>     // specify the file extension you want to import
>     declare module "*.yml" {
>       const value: any; // Add better type definitions here if desired
>       export default value;
>     }
>     ```

Reviewer 2: Nit: uppercase first letter for both comments.

Also, I think we might want to add a code block title here to specificy **where** in your project the `*.d.ts` file goes.
````suggestion
    ```ts
    // Specify the file extension you want to import
    declare module "*.yml" {
      const value: any; // Add better type definitions here if desired
      export default value;
    }
    ```
````

Author: *Appended the note to state you should place it in the `src` directory
